src/pages/AppDetails/AppDetailsPage.tsx

```diff
@@ -117,6 +125,10 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
     return istioSidecar;
   }
 
+  updateRateInterval = (rateInterval: DurationInSeconds) => {
+    this.setState({ rateInterval }, () => this.doRefresh());
```

Comment:
This seems to work, but I'm curious... how?
I mean how `this.setState({ rateInterval })` can work where I would expect the state object be more like `{ rateInterval: rateInterval }`? Some dark magic happening here?

Reply:
@jotak you can do this if the name of the variable is the same that the property in the state, typescript sometimes rocks ^^

Reply:
One man's rocks are another man's mud :)  Personally, it's this sort of optimization that just makes code confusing.

Reply:
@aljesusg thanks, good to know! But too much fancy syntactic sugar for me :-)

Reply:
@jshaughn this is where we can see we're getting old

Reply:
@jotak @jshaughn Haha... I think it is not related to oldness... Otherwise, I'd be feeling "old" about 3-5 years after writing my first line of code :P

BTW, that syntax is "new" in ES6. Read this little section:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions

Reply:
Yes, is dark magic :)